pkg/sql/logictest/testdata/logic_test/privilege_builtins

```diff
@@ -1056,3 +1056,53 @@ query B
 SELECT has_column_privilege('hcp_test'::REGCLASS, 4, 'SELECT')
 ----
 true
+
+# Regression test for #58254.
+
+user root
+
+statement ok
+DROP DATABASE IF EXISTS my_db;
+CREATE DATABASE my_db;
```

Comment:
could you also create a user-defined schema within my_db? the permissions are computed differently for those

Reply:
Done!